src/test/java/org/elasticsearch/action/IndicesRequestTests.java

```diff
@@ -432,6 +439,24 @@ public void testRefresh() {
 
         clearInterceptedActions();
         assertSameIndices(refreshRequest, refreshShardAction);
+
+        refreshRequest = new RefreshRequest("v*");
+        RefreshResponse response = internalCluster().clientNodeClient().admin().indices().refresh(refreshRequest).actionGet();
+
+        clearInterceptedActions();
+        assertThat("'v*' should not match any indices, so no shard should be returned", response.getTotalShards(), equalTo(0));
+
+        refreshRequest = new RefreshRequest("_all");
+        response = internalCluster().clientNodeClient().admin().indices().refresh(refreshRequest).actionGet();
+
+        clearInterceptedActions();
+        assertThat("'_all' should match any index, so all shards should be returned", response.getTotalShards(), equalTo(this.totalNumShards));
+
+        refreshRequest = new RefreshRequest(Strings.EMPTY_ARRAY);
+        response = internalCluster().clientNodeClient().admin().indices().refresh(refreshRequest).actionGet();
+
+        clearInterceptedActions();
+        assertThat("Empty index arrays should match any index, so all shards should be returned", response.getTotalShards(), equalTo(this.totalNumShards));
```

Comment:
wondering if these additional checks really belong here. IndicesRequestTests is a different beast, that tests indices resolution. I would probably move to a new test class that is more about testing responses and number of shards.